app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,39 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  # Remembers a user in a persistent session.
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  # Returns the user corresponding to the remember token cookie.
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
   end
 
   def logged_in?
     current_user.present?
   end
 
+  # Forgets a persistent session.
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
   def log_out
+    forget(current_user)
```

Comment:
bỏ () những nơi gọi method ko cần dùng đến đi em
```suggestion
    forget current_user
```